image/refs/interface.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package refs implements generic name-based reference access.
+package refs
+
+import (
+	"github.com/opencontainers/image-spec/specs-go"
+	"golang.org/x/net/context"
+)
+
+// Engine represents a name-based reference storage engine.
+//
+// This interface is for internal use of oci-image-tool for the time
+// being.  It is subject to change.  This notice will be removed when
+// and if the interface becomes stable.
+type Engine interface {
+
+	// Put adds a new reference to the store.  The action is idempotent;
+	// a nil return means "that descriptor is stored at NAME" without
+	// implying "because of your Put()".
+	Put(ctx context.Context, name string, descriptor *specs.Descriptor) (err error)
```

Comment:
On Tue, Aug 30, 2016 at 01:41:16PM -0700, Stephen Day wrote:
> > +	// Put adds a new reference to the store.  The action is idempotent;
> > +	// a nil return means "that descriptor is stored at NAME" without
> > +	// implying "because of your Put()".
> > +	Put(ctx context.Context, name string, descriptor *specs.Descriptor) (err error)
> 
> Do you argue with every single point made in a code review?

Nope, I picked up the ones that were clear without discussion [1].  I
just ask for motivation where the suggestion doesn't match my
preferred behavior.  In this case, I explained my reason for
preferring a pointer (save a copy [2]), and asked for your reason for
preferring pass-by-value.

[1]: https://github.com/opencontainers/image-spec/pull/159#issuecomment-243330295
[2]: https://github.com/opencontainers/image-spec/pull/159#discussion_r76726957